### PR TITLE
Add terminal emulator example using Textual

### PR DESCRIPTION
**Pull request by davidbrochart**

### CODE REVIEW

**Review by willmcgugan — COMMENTED** [review]
I would be amazing to have a terminal widget in Textual.

DM me if you ever want to brainstorm on this.

File: examples/terminal_emulator.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    async def recv(self):
+        while True:
+            message = await self.recv_queue.get()
+            cmd = message[0]
+            if cmd == "setup":
+                await self.send_queue.put(["set_size", 24, 80, 567, 573])
+            elif cmd == "stdout":
+                chars = message[1]
+                self.stream.feed(chars)
+                with CONSOLE.capture() as capture:
```
[the remark this answers is not in the document]
Comment (davidbrochart) [reply examples/terminal_emulator.py]: Thanks @willmcgugan!

**Comment by superbobry** [comment]
@davidbrochart lmk when this is ready for review.

**Comment by davidbrochart** [comment]
It is ready.

**Comment by superbobry** [comment]
Thanks!

**Pull request merged**